src/app/berkeley_migration_verifier/berkeley_migration_verifier.ml

```diff
@@ -0,0 +1,516 @@
+                     (sprintf "Cannot find %s hash ('%s') in migrated database"
+                        name hash ) )
+              else return Check.ok )
+        in
+
+        Deferred.return (Check.combine checks)
+      in
+
+      let%bind check =
+        compare_hashes
+          ~fetch_data_sql:(fun db ->
+            Sql.Mainnet.user_commands_hashes db end_global_slot )
+          ~find_element_sql:(fun hash ->
+            query_migrated_db ~f:(fun db ->
+                Sql.Berkeley.find_user_command_id_by_hash db hash ) )
+          ~name:"user_commands"
+      in
+      Test.of_check check ~name:"No missing user commands" ~idx:2 |> Test.eval ;
+
+      let%bind check =
+        compare_hashes
+          ~fetch_data_sql:(fun db ->
+            Sql.Mainnet.internal_commands_hashes db end_global_slot )
+          ~find_element_sql:(fun hash ->
+            query_migrated_db ~f:(fun db ->
+                Sql.Berkeley.find_internal_command_id_by_hash db hash ) )
+          ~name:"internal_commands"
+      in
+      Test.of_check check ~name:"No missing internal commands" ~idx:3
+      |> Test.eval ;
+
+      let%bind check =
+        assert_migrated_db_contains_only_canonical_blocks query_migrated_db
+          fork_block_height
+      in
+      Test.of_check check ~name:"Only canonical blocks in migrated archive"
+        ~idx:4
+      |> Test.eval ;
+
+      let%bind check =
+        assert_migrated_db_has_account_accessed query_migrated_db
+      in
+      Test.of_check check ~name:"Account accessed sanity" ~idx:5 |> Test.eval ;
+
+      let%bind check =
+        compare_hashes
+          ~fetch_data_sql:(fun db ->
+            Sql.Mainnet.block_hashes_only_canonical db end_global_slot )
+          ~find_element_sql:(fun hash ->
+            query_migrated_db ~f:(fun db ->
+                Sql.Berkeley.find_block_by_state_hash db hash ) )
+          ~name:"block_state_hashes"
+      in
+      Test.of_check check ~name:"No missing Blocks by state hashes" ~idx:6
+      |> Test.eval ;
+
+      let%bind check =
+        assert_all_user_and_interal_commands_account_ids_appear_in_accounts_accessed
+          query_migrated_db
+      in
+      Test.of_check check
+        ~name:
+          "All user and internal commands accounts appears in account accessed \
+           table"
+        ~idx:7
+      |> Test.eval ;
+
+      let%bind check =
+        compare_hashes
+          ~fetch_data_sql:(fun db ->
+            Sql.Mainnet.block_parent_hashes_no_orphaned db end_global_slot )
+          ~find_element_sql:(fun hash ->
+            query_migrated_db ~f:(fun db ->
+                Sql.Berkeley.find_block_by_parent_hash db hash ) )
+          ~name:"orphaned block"
+      in
+      Test.of_check check ~name:"No orphaned blocks in migrated schema" ~idx:8
+      |> Test.eval ;
+
+      let%bind check =
+        compare_hashes
+          ~fetch_data_sql:(fun db ->
+            Sql.Mainnet.ledger_hashes_no_orphaned db end_global_slot )
```

Comment:
To make sure the blocks all have the same ledger_hash, what we should do is compare the ledger hashes of the same block (with same state hash) on 2 networks.

This and the above can be combined into 1 query.